src/nvenc/nvenc_base.cpp

```diff
@@ -0,0 +1,323 @@
+        format_config.chromaFormatIDC = 3;
+      }
+      format_config.enableFillerDataInsertion = config.filler_data_insertion;
+    };
+
+    auto fill_vui = [&colorspace](auto &vui_config) {
+      vui_config.videoSignalTypePresentFlag = 1;
+      vui_config.videoFormat = NV_ENC_VUI_VIDEO_FORMAT_UNSPECIFIED;
+      vui_config.videoFullRangeFlag = colorspace.full_range;
+      vui_config.colourDescriptionPresentFlag = 1;
+      vui_config.colourPrimaries = colorspace.primaries;
+      vui_config.transferCharacteristics = colorspace.tranfer;
+      vui_config.colourMatrix = colorspace.matrix;
+    };
+
+    switch (client_config.videoFormat) {
+      case 0: {
+        // H.264
+        auto &format_config = enc_config.encodeCodecConfig.h264Config;
+        set_common_format_config(format_config);
+        if (client_config.numRefFrames > 0) {
+          format_config.maxNumRefFrames = client_config.numRefFrames;
+        }
+        else {
+          format_config.maxNumRefFrames = 5;
+        }
+        format_config.numRefL0 = NV_ENC_NUM_REF_FRAMES_1;
+        fill_vui(format_config.h264VUIParameters);
+        break;
+      }
+
+      case 1: {
+        // HEVC
+        auto &format_config = enc_config.encodeCodecConfig.hevcConfig;
+        set_common_format_config(format_config);
+        if (buffer_format == NV_ENC_BUFFER_FORMAT_YUV420_10BIT || buffer_format == NV_ENC_BUFFER_FORMAT_YUV444_10BIT) {
+          format_config.pixelBitDepthMinus8 = 2;
+        }
+        if (client_config.numRefFrames > 0) {
+          format_config.maxNumRefFramesInDPB = client_config.numRefFrames;
+        }
+        else {
+          format_config.maxNumRefFramesInDPB = 5;
+        }
+        format_config.numRefL0 = NV_ENC_NUM_REF_FRAMES_1;
+        fill_vui(format_config.hevcVUIParameters);
+        break;
+      }
+    }
+
+    init_params.encodeConfig = &enc_config;
+
+    if (nvenc->nvEncInitializeEncoder(encoder, &init_params) != NV_ENC_SUCCESS) {
+      BOOST_LOG(error) << "NvEncInitializeEncoder failed: " << nvenc->nvEncGetLastErrorString(encoder);
+      return false;
+    }
+
+    NV_ENC_CREATE_BITSTREAM_BUFFER create_bitstream_buffer = { NV_ENC_CREATE_BITSTREAM_BUFFER_VER };
+    if (nvenc->nvEncCreateBitstreamBuffer(encoder, &create_bitstream_buffer) != NV_ENC_SUCCESS) {
+      BOOST_LOG(error) << "NvEncCreateBitstreamBuffer failed: " << nvenc->nvEncGetLastErrorString(encoder);
+      return false;
+    }
+    output_bitstream = create_bitstream_buffer.bitstreamBuffer;
+
+    if (!create_input_buffer()) {
+      return false;
+    }
+
+    BOOST_LOG(info) << "Created NvENC encoder at " << quality_preset_string_from_guid(init_params.presetGUID) << (config.vbv_multiplier > 1.0f ? " with modified VDV" : "");
+
+    return true;
+  }
+
+  nvenc_encoded_frame
+  nvenc_base::encode_frame(uint64_t frame_index, bool force_idr) {
+    if (!encoder || !output_bitstream) {

```

Comment:
Can this actually happen or are you just being cautious?

Reply:
Done.